### PR TITLE
21 GUI based service conditions manipulation

### PR DESCRIPTION
**Pull request by baha-a**
[Issue #21](https://github.com/buerokratt/Service-Module/issues/21)

### CODE REVIEW

File: GUI/src/components/FlowCanvas/DeleteButton/styles.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.remove-button {
+    color: red;
+}
```
Comment (1AhmedYasser) [inline GUI/src/components/FlowCanvas/DeleteButton/styles.scss:3]: Add New Line

File: GUI/src/components/FlowCanvas/RuleStep/styles.scss [hunk 2]
```diff
@@ -0,0 +1,9 @@
+.text-input {
+    width: 70%;
+    height: 3rem;
+    background: #fff;
+    margin-top: 0;
+    border-radius: .5rem;
+    border: 1px solid black;
+    padding: .75rem;
+}
```
Comment (1AhmedYasser) [inline GUI/src/components/FlowCanvas/RuleStep/styles.scss:9]: Add New Line

File: GUI/src/components/FlowCanvas/StepDev/styles.scss [hunk 3]
```diff
@@ -0,0 +1,8 @@
+.draggable-item {
+    min-height: 40px;
+    width: 100%;
+    margin: .5rem;
+    border-radius: .5rem;
+    user-select: none;
+    padding: .75rem;
+}
```
Comment (1AhmedYasser) [inline GUI/src/components/FlowCanvas/StepDev/styles.scss:8]: Add New Line

File: GUI/src/components/FlowCanvas/SwitchStep/styles.scss [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    background: unset;
+}
+
+.opened {
+    color: '#000'
+}
+
+.closed {
+    color: #fff
+}
```
Comment (1AhmedYasser) [inline GUI/src/components/FlowCanvas/SwitchStep/styles.scss:11]: Add New Line

File: GUI/src/components/FlowCanvas/SwitchStepBody/styles.scss [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    border-radius: .5rem;
+    user-select: none;
+    padding: .75rem;
+    background: #fff;
+    border: 1px solid black;
+}
+
+.switch-body-content {
+    min-width: 30vw;
+}
```
Comment (1AhmedYasser) [inline GUI/src/components/FlowCanvas/SwitchStepBody/styles.scss:13]: Add New Line

File: GUI/src/components/FlowCanvas/TextSpace/styles.scss [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+.text-input {
+    min-height: 40px;
+    width: 100%;
+    background: #fff;
+    margin: .5rem;
+    margin-top: 0;
+    border-radius: .5rem;
+    border: 1px solid black;
+    padding: .75rem;
+}
```
Comment (1AhmedYasser) [inline GUI/src/components/FlowCanvas/TextSpace/styles.scss:15]: Add New Line

File: GUI/src/i18n/et/common.json [hunk 7]
```diff
@@ -6,6 +6,7 @@
   },
   "menu": {
     "overview": "Ülevaade",
+    "flowbuilder": "Flow Builder",
     "close": "Sulge menüü"
   }
 }
```
Comment (1AhmedYasser) [inline GUI/src/i18n/et/common.json:12]: Add New Line

**Review by 1AhmedYasser — COMMENTED** [review]
Changed Required

**Review by 1AhmedYasser — APPROVED** [review]

**Review by trevorling — COMMENTED** [review]
We should decide which is the indent for files: 2 or 4 (spaces).

What do you think about the idea that we move the `type`'s and `interface`'s to separate files?

File: GUI/src/components/FlowCanvas/TextSpace/index.tsx [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import { useDispatch } from 'react-redux'
+import { changeTextStepContentAction } from '../../../store/actions/steps';
+import { StepType } from '../../../types/reducers';
+import Track from '../../Track';
+import StepDev from '../StepDev';
+import './styles.scss'
+
+interface TextSpaceProps {
+    step: StepType
+}
```
[the remark this answers is not in the document]
Comment (baha-a) [reply GUI/src/components/FlowCanvas/TextSpace/index.tsx:11]: > We should decide which is the indent for files: 2 or 4 (spaces).

2 spaces sounds good, I've changed the code indent, and I've updated my IDE to use 2 spaces from now on.

> What do you think about the idea that we move the `type`'s and `interface`'s to separate files?

Technically, it is possible to separate the `props` interface from the component file in React, but conventionally, the `props` interfaces are defined in the same file as the component.

[the remark this answers is not in the document]
Comment (trevorling) [reply GUI/src/components/FlowCanvas/TextSpace/index.tsx:11]: Thanks for the information! :)

**Review by trevorling — APPROVED** [review]

**Review by trevorling — APPROVED** [review]

**Pull request merged**